app-web/plugins/gatsby-source-github-all/utils/plugins.js

```diff
@@ -286,6 +307,52 @@ const repositoryResourcePathPlugin = async (extension, file) => {
   return file;
 };
 
+/**
+ * creates the slug metadata property which is used by the pagePathPlugin
+ * @param {String} extension
+ * @param {Object} file
+ */
+const markdownSlugPlugin = (extension, file) => {
+  if (extension === 'md') {
+    // parse front matter
+    const data = matter(file.content, { delimiters: '---' });
+    const frontmatter = data.data;
+    let slug = frontmatter.slug || frontmatter.title;
+    slug = slugify(slug);
+    // get the current resource for the store if it exists
+    const currentResource = slugStore.get(slug);
+    try {
+      // if there is a conflict, slugstore has been configd to throw
+      slugStore.checkConflict(slug);
+    } catch (e) {
+      // throwing allows for a more detailed message.
+      const produceSummary = metadata =>
+        `Source: ${metadata.sourceName}, fileName: ${metadata.fileName}, title: ${
+          metadata.resourceTitle
+        }`;
+      const currentSummary = produceSummary(currentResource);
+      const conflictingSummary = produceSummary(file.metadata);
+      const warning = chalk`\n
+        {red WARNING from Siphon!} --- markdown file slug conflict {red.bold (slug: ${slug}) } 
+        the following markdown file ---
+        {green ${conflictingSummary}}
+        has a naming conflict in the slug that is being used to produce a gatsby page. 
+        The slug is currently in use by this markdown file ---
+        {green ${currentSummary}}
+        {cyan.bold This may cause odd issues for links to the gatsby page if not rectified.}
+        detailed stack below..
+      `;
+
+      console.error(warning);
+      console.error(e);
+    }
+    // continue to set new slug in store
+    slugStore.set(slug, file.metadata);
```

Comment:
why setting the whole metadata into slug store but not just the slug?

Reply:
oh only just for the purpose of producing the error message described above.